verified-reviews/utils/client.ts

```
@@ -0,0 +1,180 @@
+import { fetchAPI } from "../../utils/fetch.ts";
+import { Ratings, Reviews, VerifiedReviewsFullReview } from "./types.ts";
+import { Product } from "../../commerce/types.ts";
+import { ConfigVerifiedReviews } from "../mod.ts";
+import { context } from "deco/live.ts";
+
+export type ClientVerifiedReviews = ReturnType<typeof createClient>;
+
+export interface PaginationOptions {
+	count?: number;
+	offset?: number;
+	order?:
+		| "date_desc"
+		| "date_ASC"
+		| "rate_DESC"
+		| "rate_ASC"
+		| "helpfulrating_DESC";
+}
+
+const messageError = {
+	ratings:
+		"🔴⭐ Error on call ratings of Verified Review - probably unidentified product",
+	rating:
+		"🔴⭐ Error on call single rating of Verified Review - probably unidentified product",
+	fullReview:
+		"🔴⭐ Error on call Full Review of Verified Review - probably unidentified product",
+};
+
+const baseUrl = "https://awsapis3.netreviews.eu/product";
+
+export const createClient = (params: ConfigVerifiedReviews | undefined) => {
+	if (!params) return;
+
+	const { idWebsite } = params;
+
+	/** @description https://documenter.getpostman.com/view/2336519/SVzw6MK5#338f8f1b-4379-40a2-8893-080fe5234679 */
+	const rating = async ({ productId }: { productId: string }) => {
+		const payload = {
+			query: "average",
+			products: [productId],
+			idWebsite: idWebsite,
+			plateforme: "br",
+		};
+
+		try {
+			const data = await fetchAPI<Ratings>(`${baseUrl}`, {
+				method: "POST",
+				body: JSON.stringify(payload),
+			});
+
+			return Object.keys(data).length ? data : undefined;
+		} catch (error) {
+			if (context.isDeploy) {
+				console.error(messageError.rating, error);
+			} else {
```

you should throw the error again here as you're locally